File: chap13/Looting Demo/Project/Items.cs
using System;
using System.Collections.Generic;
using System.Xml;

namespace RPG
{
public class Items
{
    //keep public for easy access
    public List<Item> items;

    public Items() 
    {
        items = new List<Item>();
    }

    private string getElement(string field, ref XmlElement element)
    {
        string value = "";
        try
        {
            value = element.GetElementsByTagName(field)[0].InnerText;
        }
        catch (Exception){}
        return value;
    }

    public bool Load(string filename)
    {
        try
        {
            //open the xml file 
            XmlDocument doc = new XmlDocument();
            doc.Load(filename);
            XmlNodeList list = doc.GetElementsByTagName("item");
            foreach (XmlNode node in list)
            {
                //get next item in table
                XmlElement element = (XmlElement)node;
                Item item = new Item();

                //store fields in new Item
                item.Name = getElement("name", ref element);
                item.Description = getElement("description", ref element);
                item.DropImageFilename = getElement("dropimagefilename", ref element);
                item.InvImageFilename = getElement("invimagefilename", ref element);
                item.Category = getElement("category", ref element);
                item.Weight = Convert.ToSingle(getElement("weight", ref element));
                item.Value = Convert.ToSingle(getElement("value", ref element));
                item.AttackNumDice = Convert.ToInt32(getElement("attacknumdice", ref element));
                item.AttackDie = Convert.ToInt32(getElement("attackdie", ref element));
                item.Defense = Convert.ToInt32(getElement("defense", ref element));
                item.STR = Convert.ToInt32(getElement("STR", ref element));
                item.DEX = Convert.ToInt32(getElement("DEX", ref element));
                item.STA = Convert.ToInt32(getElement("STA", ref element));
                item.INT = Convert.ToInt32(getElement("INT", ref element));
                item.CHA = Convert.ToInt32(getElement("CHA", ref element));

                //add new item to list
                items.Add(item);
            }
        }
        catch (Exception)
        {
            return false;
        }
        return true;
    }

    public Item getItem(string name)
    {
        foreach (Item it in items)
        {
            if (it.Name == name) return it;
        }
        return null;
    }

}
}

File: chap13/Character Editor 2/Character Editor/Items.vb
Imports System.Xml

Public Class Items

    REM keep public for easy access
    Public items As List(Of Item)

    Public Sub New()
        items = New List(Of Item)
    End Sub

    Private Function getElement(ByVal field As String, ByRef element As XmlElement) As String
        Dim value As String = ""
        Try
            value = element.GetElementsByTagName(field)(0).InnerText
        Catch ex As Exception
            REM ignore error, just return empty
            Console.WriteLine(ex.Message)
        End Try
        Return value
    End Function

    Public Function Load(ByVal filename As String) As Boolean
        Try
            REM open the xml file 
            Dim doc As New XmlDocument()
            doc.Load(filename)
            Dim list As XmlNodeList = doc.GetElementsByTagName("item")
            For Each node As XmlNode In list

                REM get next item in table
                Dim element As XmlElement = node
                Dim item As New Item()

                REM store fields in new Item
                item.Name = getElement("name", element)
                item.Description = getElement("description", element)
                item.DropImageFilename = getElement("dropimagefilename", element)
                item.InvImageFilename = getElement("invimagefilename", element)
                item.Category = getElement("category", element)
                item.Weight = Convert.ToSingle(getElement("weight", element))
                item.Value = Convert.ToSingle(getElement("value", element))
                item.AttackNumDice = Convert.ToInt32(getElement("attacknumdice", element))
                item.AttackDie = Convert.ToInt32(getElement("attackdie", element))
                item.Defense = Convert.ToInt32(getElement("defense", element))
                item.STR = Convert.ToInt32(getElement("STR", element))
                item.DEX = Convert.ToInt32(getElement("DEX", element))
                item.STA = Convert.ToInt32(getElement("STA", element))
                item.INT = Convert.ToInt32(getElement("INT", element))
                item.CHA = Convert.ToInt32(getElement("CHA", element))

                REM add new item to list
                items.Add(item)
            Next
        Catch ex As Exception
            Return False
        End Try
        Return True
    End Function

    Public Function getItem(ByVal name As String) As Item
        For Each it As Item In items
            If it.Name = name Then
                Return it
            End If
        Next
        Return Nothing
    End Function

End Class
